docker/pulsar-all/pom.xml

```diff
@@ -119,57 +111,73 @@
-              </execution>
-              <execution>
-                <id>add-no-repo-and-latest</id>
-                <goals>
-                  <goal>tag</goal>
+                  <goal>exec</goal>
                 </goals>
                 <configuration>
-                  <repository>pulsar-all</repository>
-                  <tag>latest</tag>
+                  <environmentVariables>
+                    <DOCKER_BUILDKIT>1</DOCKER_BUILDKIT>
+                  </environmentVariables>
+                  <executable>docker</executable>
+                  <workingDirectory>${project.basedir}</workingDirectory>
+                  <arguments>
+                    <argument>build</argument>
+                    <argument>--no-cache</argument>
+                    <argument>--build-arg</argument>
+                    <argument>PULSAR_TARBALL=target/pulsar-server-distribution-${project.version}-bin.tar.gz</argument>
+                    <argument>-t</argument>
+                    <argument>pulsar-all:${project.version}</argument>
+                    <argument>-t</argument>
+                    <argument>pulsar-all:latest</argument>
+                    <argument>-t</argument>
+                    <argument>${docker.organization}/pulsar-all:${project.version}</argument>
+                    <argument>-t</argument>
+                    <argument>${docker.organization}/pulsar-all:latest</argument>
+                    <argument>.</argument>
+                  </arguments>
                 </configuration>
               </execution>
               <execution>
-                <id>tag-and-push-latest</id>
+                <id>docker-push</id>
+                <phase>deploy</phase>
                 <goals>
-                  <goal>tag</goal>
-                  <goal>push</goal>
+                  <goal>exec</goal>
                 </goals>
                 <configuration>
-                  <repository>${docker.organization}/pulsar-all</repository>
-                  <tag>latest</tag>
+                  <environmentVariables>
+                    <DOCKER_BUILDKIT>1</DOCKER_BUILDKIT>
+                  </environmentVariables>
+                  <executable>docker</executable>
+                  <workingDirectory>${project.basedir}</workingDirectory>
+                  <arguments>
+                    <argument>buildx</argument>
+                    <argument>build</argument>
+                    <argument>--no-cache</argument>
+                    <argument>--platform=linux/amd64,linux/arm64</argument>
+                    <argument>--build-arg</argument>
+                    <argument>
+                      PULSAR_IO_DIR=target/apache-pulsar-io-connectors-${project.version}-bin
+                    </argument>
+                    <argument>
+                      PULSAR_OFFLOADER_TARBALL=target/pulsar-offloader-distribution-${project.version}-bin.tar.gz
+                    </argument>
+                    <argument>--push</argument>
```

Comment:
I think this `--push` might try to push the image to the DockerHub repo instead of local. Do we want to `push` the image to local or DockerHub here?